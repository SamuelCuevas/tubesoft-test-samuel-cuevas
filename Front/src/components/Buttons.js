import React from 'react';


function Buttons({ start, status, pause, stop, resume, clickAlert }) {

  return (
      <div className="buttons">
          { (status === 0)? 
            <button onClick={ start }>Start</button>
            : <button onClick={ resume }>Resume</button>
          }
          
          { (status === 1 )?
              <button onClick={ pause }>Pause</button>
            : <button>Pause</button>
          }

          { (status === 1 || status === 2)?
            <button onClick={ stop }>Stop</button>
            : <button>Stop</button>
          }
          
      </div>
  );
}
export default Buttons;