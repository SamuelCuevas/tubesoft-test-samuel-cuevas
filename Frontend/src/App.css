body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

:root {
    --background: rgb(2,0,36);
    --container: #353535;
    --text-color: white;
    --screen: #353535;
    --btn-hover: black;
    --btn-hover-text:#00d4ff;
    --shadow: rgba(59, 59, 59, 0.75);
}

span {
  font-weight: 300px;
  font-size: 100px;
  color: var(--text-color);
}

.App {
  background: var(--background);
  background: linear-gradient(150deg, rgba(2,0,36,1) 0%, rgba(9,86,121,1) 53%, rgba(0,212,255,1) 100%);;
  width: 100%;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
}

.time {
  background-color: var(--container);
  width: 650px;
  height: 250px;
  border-radius: 20px;
  box-shadow: 11px 17px 22px 0px var(--shadow);
}

.time.chronometer {
  color: var(--text-color);
  background-color: var(--screen);
  margin: 6px;
  text-align: center;
}

.buttons {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  text-align: center;
}

.buttons>button{    
  display: block;
    background: none;
    border: none;
    outline: inherit;
    cursor: pointer;

    height: 80px;
    border-radius: 50%;
    font-size: 50px;
    color: var(--text-color);
}

button:hover{
  transform: scale(1.2);
  transition: 0.5s;
  background-color: var(--btn-hover);
  color: var(--btn-hover-text);
}


