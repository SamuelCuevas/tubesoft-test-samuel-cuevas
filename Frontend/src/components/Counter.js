import React from 'react';


function Counter({ time }) {

  const h = () => {
    if(time.h === 0){
      return '';
    }else {
      return <span>{ (time.h >= 10 )? time.h : "0"+ time.h }:</span>
    }
 }

  return (
      <div>
          {h()}
          <span>{ (time.m >= 10 )? time.m : "0"+ time.m }:</span>
          <span>{ (time.s >= 10 )? time.s : "0"+ time.s }:</span>
          <span>{ (time.ms >= 10 )? time.ms : "0"+ time.ms } </span>
      </div>
  );
}
export default Counter;