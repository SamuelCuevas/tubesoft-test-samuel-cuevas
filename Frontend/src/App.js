import { useState } from 'react';
import Buttons from './components/Buttons';
import Counter from './components/Counter';
import { handleFetch } from './helpers/fetch';

import './App.css';

function App() {

  const [time, setTime] = useState({ ms:0, s:0, m:0, h:0 });
  const [interv, setInterv] = useState();
  const [status, setStatus] = useState(0);
  let { ms, s, m, h } = time;

  const handleStart = () => {
    run();
    setInterv(setInterval( run, 10 ));
    setStatus(1);
  };

  const handleResume = () => {
    if( status === 2 ){ 
        run();
        setInterv(setInterval( run, 10 ));
        setStatus(1);
    }
  };

  const handleStop = () => {
    handleFetch(h,m,s,ms);
    clearInterval( interv );
    setStatus(0);
    setTime({ ms:0, s:0, m:0, h:0 });
  };

  const handlePause = () => {
    clearInterval( interv );
    setStatus(2);
  };

  let updatedMs = ms, updatedS = s, updatedM = m, updatedH = h;

  const run = () => {

    if( updatedM === 60 ) {
        updatedH++;
        updatedM = 0;
    }

    if( updatedS === 60 ) {
        updatedM++;
        updatedS = 0;
    }

    if( updatedMs === 100 ) {
        updatedS++;
        updatedMs = 0;
    }

    updatedMs++;
    return setTime({ ms:updatedMs, s:updatedS, m:updatedM, h:updatedH });

  };

  return (
      <div className="App">
          <div className="time">
              <div className="chronometer">
                  <Counter time={ time } />
                  <Buttons status={ status } start={ handleStart } stop={ handleStop } pause={ handlePause } resume={ handleResume } />
              </div>
          </div>
      </div>
  );
}
export default App;
