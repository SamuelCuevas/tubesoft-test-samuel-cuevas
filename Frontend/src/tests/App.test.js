import React from 'react';
import { shallow, mount, render } from 'enzyme';

// import Foo from '../Foo';
import App from '../App';
import Counter from '../components/Counter';
import Buttons from '../components/Buttons';

describe('Tests in App.js', () => {

    const t = {h:0,m:0,s:0,ms:0};

    const wrapper = shallow( <App />);
    const buttons = shallow( <Buttons/> );
    const time = shallow( <Counter time={ t } /> );

    test('should Component <App/> correctly', () => {

        expect(wrapper).toMatchSnapshot();
    })

    test('should start the chronometer', () => {
       
        buttons.find('button').at(0).simulate('click');
        const counter = time.find('#time').text().trim();

        expect( counter ).toBe('00:00:00');
        
    });

});