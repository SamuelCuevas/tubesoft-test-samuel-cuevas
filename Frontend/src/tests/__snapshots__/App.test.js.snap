// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in App.js should Component <App/> correctly 1`] = `
<div
  className="App"
>
  <div
    className="time"
  >
    <div
      className="chronometer"
    >
      <Counter
        time={
          Object {
            "h": 0,
            "m": 0,
            "ms": 0,
            "s": 0,
          }
        }
      />
      <Buttons
        pause={[Function]}
        resume={[Function]}
        start={[Function]}
        status={0}
        stop={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`Tests in App.js should Component <App/> correctly 1`] = `
<div
  className="App"
>
  <div
    className="time"
  >
    <div
      className="chronometer"
    >
      <Counter
        time={
          Object {
            "h": 0,
            "m": 0,
            "ms": 0,
            "s": 0,
          }
        }
      />
      <Buttons
        pause={[Function]}
        resume={[Function]}
        start={[Function]}
        status={0}
        stop={[Function]}
      />
    </div>
  </div>
</div>
`;
